starknet-e2e-test/ethereum_settlement.rs

```diff
@@ -1,98 +1,54 @@
-use starknet_e2e_test::ethereum_sandbox::EthereumSandbox;
-use starknet_e2e_test::starknet_contract::{InitData, StarknetContract};
-use tempfile::TempDir;
+use starknet_e2e_test::starknet_sovereign::StarknetSovereign;
 use test_context::{test_context, AsyncTestContext};
 use tokio::time::sleep;
 
 struct Context {
-    pub madara_path: TempDir,
-    pub sandbox: EthereumSandbox,
-    pub starknet_contract: StarknetContract,
+    pub madara_temp_dir: MadaraTempDir,
+    pub starknet_sovereign: StarknetSovereign,
 }
 
 impl Context {
-    pub async fn initialize(&self, data: InitData) {
-        self.starknet_contract.initialize(&self.sandbox, data).await;
-    }
-
     pub async fn launch_madara(&self) -> MadaraRunner {
-        MadaraRunner::new(Some(Settlement::Ethereum), Some(self.madara_path.path().to_path_buf())).await
+        let settlement_conf = self.starknet_sovereign.create_settlement_conf(self.madara_temp_dir.data_path()).await;
+        MadaraRunner::new(MadaraArgs {
+            settlement: Some(Settlement::Ethereum),
+            settlement_conf: Some(settlement_conf),
+            base_path: Some(self.madara_temp_dir.base_path()),
+        })
+        .await
     }
 
     pub async fn read_state(&self) -> StarknetState {
-        let client = StarknetContractClient::new(self.starknet_contract.address(), self.sandbox.client());
+        let client = StarknetContractClient::new(self.starknet_sovereign.address(), self.starknet_sovereign.client());
         SettlementProvider::<Block>::get_state(&client).await.expect("Failed to get state")
     }
 }
 
 #[async_trait]
 impl AsyncTestContext for Context {
     async fn setup() -> Self {
-        let sandbox = EthereumSandbox::new();
-        let starknet_contract = StarknetContract::deploy(&sandbox).await;
-
-        let madara_path = TempDir::with_prefix("madara").expect("Failed to create Madara path");
-        let config_dir = madara_path.path().join("chains/dev"); // data path
-        create_dir_all(&config_dir).unwrap();
-
-        let config = EthereumConfig {
-            http_provider: sandbox.client().provider().url().to_string(),
-            core_contracts: hex_str_from_bytes::<20, true>(starknet_contract.address().0),
-            chain_id: sandbox.client().get_chainid().await.expect("Failed to get sandbox chain ID").as_u64(),
-            poll_interval_ms: Some(10u64), // Default is 7s, we need to speed things up
-            ..Default::default()
-        };
-
-        let config_file = File::create(config_dir.join("eth-config.json")).expect("Failed to open file for writing");
-        serde_json::to_writer(config_file, &config).expect("Failed to write eth config");
-
-        Self { madara_path, sandbox, starknet_contract }
+        let starknet_sovereign = StarknetSovereign::deploy().await;
+        let madara_temp_dir = MadaraTempDir::default();
+        Self { madara_temp_dir, starknet_sovereign }
     }
 
     async fn teardown(self) {
```

Comment:
I don't see this method being called anywhere in the tests.
You could maybe implement `Drop` on `Context` and call `teardown` there

Reply:
Oh mb. This is `AsyncTestContext` trait. They are automatically called by the test runner